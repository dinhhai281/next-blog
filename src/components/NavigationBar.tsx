import {
  Box,
  Button,
  ButtonGroup,
  Container,
  Popover,
  PopoverBody,
  PopoverCloseButton,
  PopoverContent,
  PopoverFooter,
  PopoverTrigger,
  Text,
} from '@chakra-ui/react';
import React, { FC } from 'react';
import { DropdownMenu } from './DropdownMenu';

export interface NavigationBarProps {
  title: string;
}

export const NavigationBar: FC<NavigationBarProps> = ({ title }) => {
  return (
    <Box bg='whiteAlpha.900' py={6}>
      <Container maxW={['full', '3xl']} h={10} d='flex' alignItems='center'>
        <Text fontSize='4xl' color='blue.600' as='em' mr={6}>
          {title}
        </Text>
        <DropdownMenu items={['Programming', 'Cooking', 'Gaming']}>Sections</DropdownMenu>
      </Container>
    </Box>
  );
};
