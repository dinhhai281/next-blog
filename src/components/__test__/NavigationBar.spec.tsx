import { render } from '@testing-library/react';
import React from 'react';

import { NavigationBar } from '../NavigationBar';

describe('NavigationBar', () => {
  const renderComponent = () => render(<NavigationBar title='title' />);

  describe('when render', () => {
    it('should match snapshot', () => {
      const { asFragment } = renderComponent();
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
