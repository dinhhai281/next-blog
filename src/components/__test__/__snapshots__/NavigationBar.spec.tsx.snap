// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar when render should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-frax8c"
  >
    <div
      class="chakra-container css-fn4net"
    >
      <em
        class="chakra-text css-1i56fwb"
      >
        title
      </em>
      <button
        aria-controls="popover-content-2"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="chakra-button css-oqfhsh"
        id="popover-trigger-2"
        type="button"
      >
        <p
          class="chakra-text css-191iiyz"
        >
          Sections
        </p>
        <span
          class="chakra-button__icon css-6eepfv"
        >
          <div
            aria-hidden="true"
            class="css-0"
            focusable="false"
          >
            <svg
              class="chakra-icon css-w31a5m"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="currentColor"
              />
            </svg>
          </div>
        </span>
      </button>
      <div
        class="css-0"
        style="visibility: hidden; position: absolute; left: 0px; top: 0px;"
      >
        <section
          aria-describedby="popover-body-2"
          class="chakra-popover__content css-1ab1jlb"
          id="popover-content-2"
          role="tooltip"
          style="transform-origin: top center; opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
          tabindex="-1"
        >
          <div
            class="chakra-popover__body css-0"
            id="popover-body-2"
          >
            <button
              class="chakra-button css-1cgpak6"
              type="button"
            >
              Programming
            </button>
            <button
              class="chakra-button css-1cgpak6"
              type="button"
            >
              Cooking
            </button>
            <button
              class="chakra-button css-1cgpak6"
              type="button"
            >
              Gaming
            </button>
          </div>
        </section>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
